### ISSUE TITLE
code not  auto complate

### ISSUE THREAD

**Issue by lijunen2009** [dart-lang/sdk#36380]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA IU-183.6156.11`
- `2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36`
- `IU-183.6156.11, JRE 1.8.0_152-release-1343-b28x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.14, screens 1920x1080`

## Exception
```
Dart analysis server, SDK version 2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36, server version 1.23.0, error: Analysis failed: /Users/lijunen/Code/fitness_app/test/widget_test.dart context: exception_20190329_132125_094
Bad state: Too many elements
#0      List.single (dart:core/runtime/libgrowable_array.dart:227:5)
#1      ExprBuilder.build (package:analyzer/src/summary/expr_builder.dart:293:18)
#2      ExprTypeComputer.compute (package:analyzer/src/summary/link.dart:2526:29)
#3      TypeInferenceNode.evaluate (package:analyzer/src/summary/link.dart:5286:36)
#4      TypeInferenceDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:5136:7)
#5      DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2217:13)
#6      DependencyWalker.walk (package:analyzer/src/summary/link.dart:2236:18)
#7      VariableElementForLink.inferredType (package:analyzer/src/summary/link.dart:5562:47)
#8      TopLevelVariableElementForLink.link (package:analyzer/src/summary/link.dart:5121:48)
#9      CompilationUnitElementInBuildUnit.link (package:analyzer/src/summary/link.dart:1640:16)
#10     LibraryElementInBuildUnit.link (package:analyzer/src/summary/link.dart:3810:12)
#11     LibraryCycleNode.link (package:analyzer/src/summary/link.dart:3516:15)
#12     LibraryCycleDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:3456:7)
#13     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2217:13)
#14     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#15     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#16     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#17     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#18     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#19     DependencyWalker.walk (package:analyzer/src/summary/link.dart:2236:18)
#20     LibraryCycleForLink.ensureLinked (package:analyzer/src/summary/link.dart:3489:42)
#21     Linker.link (package:analyzer/src/summary/link.dart:4012:35)
#22     _relink (package:analyzer/src/summary/link.dart:305:57)
#23     link (package:analyzer/src/summary/link.dart:123:3)
#24     LibraryContext.load.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:170:25)
#25     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#26     LibraryContext.load (package:analyzer/src/dart/analysis/library_context.dart:169:12)
#27     new LibraryContext (package:analyzer/src/dart/analysis/library_context.dart:67:5)
#28     AnalysisDriver._createLibraryContext (package:analyzer/src/dart/analysis/driver.dart:1612:29)
#29     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1426:30)
#30     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#31     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1416:20)
#32     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1236:33)
<asynchronous suspension>
#33     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2145:24)
<asynchronous suspension>
#34     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:2075:5)
#35     new AnalysisServer (package:analysis_server/src/analysis_server.dart:213:29)
#36     SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:86:26)
#37     StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:37:18)
#38     Driver.startAnalysisServer.<anonymous closure> (package:analysis_server/src/server/driver.dart:526:21)
#39     _rootRun (dart:async/zone.dart:1124:13)
#40     _CustomZone.run (dart:async/zone.dart:1021:19)
#41     _runZoned (dart:async/zone.dart:1516:10)
#42     runZoned (dart:async/zone.dart:1463:12)
#43     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:605:12)
#44     Driver.startAnalysisServer (package:analysis_server/src/server/driver.dart:524:7)
#45     Driver.start (package:analysis_server/src/server/driver.dart:427:7)
#46     main (file:///b/rr/tmpgY2JrN/w/src/third_party/dart/pkg/analysis_server/bin/server.dart:12:11)
#47     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#48     main (file:///b/rr/tmpgY2JrN/w/src/third_party/dart/pkg/analysis_server/bin/server.dart:10:10)
#49     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:298:32)
#50     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

For additional log information, please append the contents of
file:///private/var/folders/9q/hsl3yn2s6fvbcvxx62jgb86w0000gn/T/report9.txt.

**Comment by bwilkerson** [comment]
@scheglov

**Comment by simolus3** [comment]
I'm think I'm getting the same error, it occurs when opening [this project](https://github.com/simolus3/moor) in IntelliJ. When I directly run `dartanalyzer .` on the affected dir, no error occurs. It's also working with VS Code, but the analyzer crashes every time I open the project in IntelliJ.

## Version information

- `IDEA IU-183.5912.21`
- `2.2.0`
- `IU-183.5912.21, JRE 1.8.0_152-release-1343-b28x64 JetBrains s.r.o, OS Linux(amd64) v5.0.5-1-MANJARO, screens 1920x1080, 1920x1080`

## Exception
```
Dart analysis server, SDK version 2.2.0, server version 1.23.0, error: Analysis failed: /home/simon/IdeaProjects/moor/moor/test/expressions/variable_test.dart context: exception_20190330_161342_831
Bad state: Too many elements
#0      List.single (dart:core/runtime/libgrowable_array.dart:227:5)
#1      ExprBuilder.build (package:analyzer/src/summary/expr_builder.dart:293:18)
#2      ExprTypeComputer.compute (package:analyzer/src/summary/link.dart:2526:29)
#3      TypeInferenceNode.evaluate (package:analyzer/src/summary/link.dart:5286:36)
#4      TypeInferenceDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:5136:7)
#5      DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2217:13)
#6      DependencyWalker.walk (package:analyzer/src/summary/link.dart:2236:18)
#7      VariableElementForLink.inferredType (package:analyzer/src/summary/link.dart:5562:47)
#8      TopLevelVariableElementForLink.link (package:analyzer/src/summary/link.dart:5121:48)
#9      CompilationUnitElementInBuildUnit.link (package:analyzer/src/summary/link.dart:1640:16)
#10     LibraryElementInBuildUnit.link (package:analyzer/src/summary/link.dart:3810:12)
#11     LibraryCycleNode.link (package:analyzer/src/summary/link.dart:3516:15)
#12     LibraryCycleDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:3456:7)
#13     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2217:13)
#14     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#15     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#16     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#17     DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:2183:24)
#18     DependencyWalker.walk (package:analyzer/src/summary/link.dart:2236:18)
#19     LibraryCycleForLink.ensureLinked (package:analyzer/src/summary/link.dart:3489:42)
#20     Linker.link (package:analyzer/src/summary/link.dart:4012:35)
#21     _relink (package:analyzer/src/summary/link.dart:305:57)
#22     link (package:analyzer/src/summary/link.dart:123:3)
#23     LibraryContext.load.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:170:25)
#24     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#25     LibraryContext.load (package:analyzer/src/dart/analysis/library_context.dart:169:12)
#26     new LibraryContext (package:analyzer/src/dart/analysis/library_context.dart:67:5)
#27     AnalysisDriver._createLibraryContext (package:analyzer/src/dart/analysis/driver.dart:1612:29)
#28     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1426:30)
#29     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#30     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1416:20)
#31     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1236:33)
<asynchronous suspension>
#32     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2145:24)
<asynchronous suspension>
#33     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:2075:5)
#34     new AnalysisServer (package:analysis_server/src/analysis_server.dart:213:29)
#35     SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:86:26)
#36     StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:37:18)
#37     Driver.startAnalysisServer.<anonymous closure> (package:analysis_server/src/server/driver.dart:526:21)
#38     _rootRun (dart:async/zone.dart:1124:13)
#39     _CustomZone.run (dart:async/zone.dart:1021:19)
#40     _runZoned (dart:async/zone.dart:1516:10)
#41     runZoned (dart:async/zone.dart:1463:12)
#42     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:605:12)
#43     Driver.startAnalysisServer (package:analysis_server/src/server/driver.dart:524:7)
#44     Driver.start (package:analysis_server/src/server/driver.dart:427:7)
#45     main (file:///b/s/w/ir/cache/builder/sdk/pkg/analysis_server/bin/server.dart:12:11)
#46     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#47     main (file:///b/s/w/ir/cache/builder/sdk/pkg/analysis_server/bin/server.dart:10:10)
#48     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:298:32)
#49     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```
```
1553958820771 started analysis server:
  /opt/dart-sdk/bin/dart /opt/dart-sdk/bin/snapshots/analysis_server.dart.snapshot --client-id=IntelliJ-IDEA --client-version=IU-183.5912.21 --useAnalysisHighlight2
1553958821421 <= {"event":"server.connected","params":{"version":"1.23.0","pid":10298,"sessionId":""}}
1553958821457 <= {"id":"0","result":{"version":"1.23.0"}}
1553958821466 => {"id":"1","method":"server.setSubscriptions","params":{"subscriptions":["STATUS"]},"clientRequestTime":1553958820775}
1553958821466 => {"id":"2","method":"analysis.updateOptions","params":{"options":{"enableAsync":true,"enableDeferredLoading":true,"enableEnums":true,"enableNullAwareOperators":true,"enableSuperMixins"
1553958821466 => {"id":"3","method":"analysis.setAnalysisRoots","params":{"included":["/home/simon/IdeaProjects/moor/moor_generator","/home/simon/IdeaProjects/moor/moor","/home/simon/IdeaProjects/moor
1553958821487 <= {"id":"1"}
1553958821487 <= {"id":"2"}
1553958821931 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor/analysis_options.yaml","errors":[]}}
1553958821934 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor/pubspec.yaml","errors":[]}}
1553958822073 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor_flutter/pubspec.yaml","errors":[]}}
1553958822119 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor_flutter/example/pubspec.yaml","errors":[]}}
1553958822168 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor_generator/analysis_options.yaml","errors":[]}}
1553958822169 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor_generator/pubspec.yaml","errors":[]}}
1553958822210 <= {"id":"3"}
1553958822215 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1553958822302 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor/example/example.dart","errors":[]}}
1553958822346 <= {"event":"analysis.errors","params":{"file":"/home/simon/IdeaProjects/moor/moor/example/example.g.dart","errors":[]}}
1553958822849 <= {"event":"server.error","params":{"isFatal":false,"message":"Analysis failed: /home/simon/IdeaProjects/moor/moor/test/expressions/variable_test.dart context: exception_20190330_161342
```

**Comment by scheglov** [comment]
@simolus3 I was not able to reproduce the issue when opened `moor` in IntelliJ. It might be because the file that caused the crash was not in the state it is in the repo now, or analyzer was changed since then.

There might be a crash log file `exception_20190330_161342` in `~/.dartServer/.analysis-driver`. Could you send it to me (scheglov@google.com)? Please note, that this file contains the transitive closure of all files that are required for analyzing the file caused the crash.

**Comment by scheglov** [comment]
Thanks, I was able to reproduce using older version of Dart SDK (older for me, but actually 2.2.0 (ref e51bdfb), so the latest released).

**Comment by scheglov** [comment]
TL;DR: remove `~/.dartServer/.analysis-driver`, and do this every time when you switch between Dart SDK versions, for now.

I suspect that this issue is caused by a cache versioning mistake in analyzer / analysis_server.

When I run analysis without any caching (using `MemoryByteStore`) and `ContextLocator` API, there are no exceptions.

When I run AnalysisServer manually, outside of IDE (to be able to debug it), and give it the same set of analysis roots via `stdin`, including `/Users/scheglov/tmp/moor`, and clear , it also works fine, without exceptions.

But when I open something using `package:front_end` using newer version of AnalysisServer, and then run another AnalysisServer manually on `/Users/scheglov/tmp/moor`, and debug it, I see a constant in `package:front_end/src/base/libraries_specification.dart` where the operation does not make sense - `pushInt` where the default value is `const <String, TargetLibrariesSpecification>{}`, so I'd expect `makeTypedMap`. So, I think we missed updating `DATA_VERSION` when incompatibly changed `UnlinkedExprAssignOperator` in`idl.dart`, although I cannot prove this with a specific PR.

https://dart-review.googlesource.com/c/sdk/+/98286

**Comment by lijunen2009** [comment]
删除~/.dartServer/.analysis-driver    ok
